test/models/role_test.rb

```diff
@@ -36,6 +36,15 @@ class RoleTest < ActiveSupport::TestCase
     role.must_be :valid?
   end
 
+  it "Should delete role who has users" do
```

Comment:
nitpick - lowercase should. can be fixed on merge. Also `FactoryGirl.build` is preferable for creating the User here, it'll prove the point just the same (and the relationship need not to be in db)